### PR TITLE
Creating talks

### PR DESCRIPTION
**Pull request by clairgriffiths**

### CODE REVIEW

**Comment by Najaf** [comment]
Speaker name and video URL don't look like they're being set.
screen shot 2016-05-03 at 16 46 55

I suspect this is because Contentful uses `camelCase` instead of `snake_case` for hash keys.

**Comment by Najaf** [comment]
Also, I'm getting this when I run `Person.all`:

```
[12] pry(main)> Person.all
NoMethodError: undefined method `talks=' for #<HbsContent::Person:0x007fedc3312cf8>
from /Users/najafali/Code/hbs/hbs-content/lib/hbs_content/contentful_data_provider.rb:97:in `block in convert'
```

The changeset doesn't appear to have `attr_accessor :talks` in `Person`.

**Comment by clairgriffiths** [comment]
On it.

**Comment by clairgriffiths** [comment]
@Najaf Fixed both of those, would you mind doing another check? Thanks

File: lib/hbs_content/contentful_data_provider.rb [hunk 1]
```diff
@@ -88,9 +90,12 @@ def convert
           p.linkedin_url      = fields[:linkedInUrl]
           p.github_url        = fields[:githubUrl]
           p.twitter_url       = fields[:twitterUrl]
-	  p.slug              = fields[:slug]
+          p.slug              = fields[:slug]
           p.type              = :person
 
+          # Deal with nil talks
+          fields[:talks] == nil ? p.talks = nil : p.talks  = fields[:talks].map { |i| MetaConverter.new(i).convert }
+
```
Comment (Najaf) [inline lib/hbs_content/contentful_data_provider.rb]: @clairgriffiths  Couple of things about this:

Your usage of the ternary operator is a bit unorthodox here, so might surprise some developers. Normally I see it used during assignment, so `p.talks` would be all the way over to the left and the branches would contain values. 

Something like this:
```
p.talks = fields[:talks].nil? ? nil : fields[:talks].map { |i| MetaConverter.new(i).convert }
```

In any case, I think the return value of nil when there are no talks is a bit surprising for users of this code. When the person object has no talks, I think most developers would expect to get an empty array back here. Do you agree?

You could achieve this with code like this:

```
p.talks = Array(fields[:talks]).map { |i| MetaConverter.new(i).convert }
```

Let me know what you think.

Comment (clairgriffiths) [inline lib/hbs_content/contentful_data_provider.rb]: Sure, I hadn't noticed it anywhere else in your code at all so I wasn't sure (I think I've got into the habit from doing lots of Code Challenges, they seem to like the ternary operator)

I agree that an empty array would be better than nil as well, I hadn't thought it through like that (I was just trying to replicate what was returned in the JSON hash).

I'll make those changes and then merge, thanks!

Comment (Najaf) [inline lib/hbs_content/contentful_data_provider.rb]: 👍

**Pull request merged**